File: core/src/main/java/top/fksoft/server/http/config/NetworkInfo.java
package top.fksoft.server.http.config;

public class NetworkInfo {
    private String hostName = null;
    private String address = null;
    private int port = -1;

    public NetworkInfo(String address, int port) {
        update(address,port);
    }

    public NetworkInfo() {
    }

    public void update(String address, int port) {
        this.address = address;
        this.port = port;
    }
    public void update(NetworkInfo info) {
        this.address = info.address;
        this.port = info.port;
        this.hostName = info.hostName;
    }
    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public int getPort() {
        return port;
    }

    public void setPort(int port) {
        this.port = port;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        NetworkInfo that = (NetworkInfo) o;
        if (port != that.port) return false;
        return address.equals(that.address);
    }

    @Override
    public int hashCode() {
        int result = address.hashCode();
        result = 31 * result + port;
        return result;
    }

    public void setHostName(String hostName) {
        this.hostName = hostName;
    }

    public String getHostName() {
        return hostName == null?address:hostName;
    }

    @Override
    public String toString() {
        return getHostName() + ":" + port;
    }
}

File: core/src/main/java/top/fksoft/server/http/config/NetworkInfo.kt
package top.fksoft.server.http.config

class NetworkInfo {
    private var hostName: String? = null
    var address: String? = null
    var port = -1

    constructor(address: String, port: Int) {
        update(address, port)
    }

    constructor() {}

    fun update(address: String, port: Int) {
        this.address = address
        this.port = port
    }

    fun update(info: NetworkInfo) {
        this.address = info.address
        this.port = info.port
        this.hostName = info.hostName
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as NetworkInfo?
        return if (port != that!!.port) false else address == that.address
    }

    override fun hashCode(): Int {
        var result = address!!.hashCode()
        result = 31 * result + port
        return result
    }

    fun setHostName(hostName: String) {
        this.hostName = hostName
    }

    fun getHostName(): String? {
        return if (hostName == null) address else hostName
    }

    override fun toString(): String {
        return getHostName() + ":" + port
    }
}
